### PR TITLE
chore(shorebird_code_push): Release 1.0.0, change downloadUpdate to downloadUpdateIfAvailable

### PR DESCRIPTION
**Pull request by bryanoltman**
Updates `downloadUpdate` method to `downloadUpdateIfAvailable` to better reflect the underlying functionality. 

Also updates/restructures the readme.

Bumps version to 1.0.0 because this is a breaking change to the API.

### CODE REVIEW

**Comment by eseidel** [comment]
I thought dartdocs were auto-generated by pub, I'm surprised they need to be checked in?

File: shorebird_code_push/README.md [hunk 1]
```diff
@@ -2,12 +2,30 @@
 
 [![License: MIT][license_badge]][license_link]
```
Comment (eseidel) [inline shorebird_code_push/README.md:3]: I'd add a discord badge here while you're at it. :)

File: shorebird_code_push/pubspec.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: shorebird_code_push
 description: Check for and download Shorebird code push updates from your app.
-version: 0.1.3
+version: 1.0.0
```
Comment (eseidel) [inline shorebird_code_push/pubspec.yaml:3]: I suspect we're going to make lots more breaking changes to the APIs soon, but I don't really care what the version number is.

**Review by eseidel — APPROVED** [review]

File: shorebird_code_push/pubspec.yaml (hunk 2 above)
Comment (bryanoltman) [reply shorebird_code_push/pubspec.yaml:3]: Yeah, this is just a) being Correct™ with semver and b) removing doubt that it's ready for use

File: shorebird_code_push/README.md (hunk 1 above)
Comment (bryanoltman) [reply shorebird_code_push/README.md:4]: Done.

**Pull request merged**